src/client.cpp

```diff
@@ -473,7 +473,7 @@ void UpdatePort(unsigned int port, bool connected, const game_controller* contro
 
   const unsigned int device = CInputManager::Get().GetDevice(port);
 
-  if (CLIENT)
+  if (CLIENT && (int)port >= 0)
```

Comment:
shouldn't cast here. If port is negative, it shouldn't be passed as an unsigned value. Either we change the parameter to negative, or we mandate that UpdatePort() is only for joysticks